src/components/accounts/recovery_setup/SetupRecoveryMethod.js

```diff
@@ -53,7 +53,27 @@ class SetupRecoveryMethod extends Component {
         email: '',
         success: false,
         emailInvalid: false,
-        phoneInvalid: false
+        phoneInvalid: false,
+        activeMethods: []
+    }
+
+    componentDidMount() {
+        const {loadRecoveryMethods, accountId, router } = this.props;
+        const { activeMethods, method } = router.location;
```

Comment:
not sure I understand this, why are we passing `activeMethods` through `router.location`?

Reply:
(I mean this seems like something that should be taken from Redux store)

Reply:
I passed `activeMethods` through router to avoid having to 'check' which methods are active when user lands on `/setup-recovery`. Just to speed up the UX and avoid loading.

Reply:
Why not cache it using Redux store instead?